### PR TITLE
Feat: use WS for wallet token balance updates

### PR DESCRIPTION
**Pull request by LukasDeco**
- Use WS to subscribe to ATAs of the user's wallet
- Update token balances from WS events
- Remove all re-fetching code like after minting conditionals and placing orders

### CODE REVIEW

File: contexts/BalancesContext.tsx [hunk 1]
```diff
@@ -71,6 +73,46 @@ export function BalancesProvider({
...
+    const tokenAccountPubkeys = await findTokenAccountsByOwner(owner);
+
+    for (const pubkey of tokenAccountPubkeys) {
+      connection.onAccountChange(pubkey, (accountInfo: AccountInfo<Buffer>) => {
+        const accountData = AccountLayout.decode(accountInfo.data);
+
+        const dividedTokenAmount = new BN(accountData.amount) / new BN(LAMPORTS_PER_SOL);
+        const tokenVal: TokenAmount = {
+          amount: dividedTokenAmount.toString(),
+          decimals: 9,
```
Comment (R-K-H) [inline contexts/BalancesContext.tsx]: So this works for META, but not USDC as USDC has 6 decimals.

File: contexts/BalancesContext.tsx [hunk 2]
```diff
@@ -71,6 +73,46 @@ export function BalancesProvider({
...
+
+  async function subscribeToTokenBalances() {
+    if (!owner) return;
+    const tokenAccountPubkeys = await findTokenAccountsByOwner(owner);
+
+    for (const pubkey of tokenAccountPubkeys) {
+      connection.onAccountChange(pubkey, (accountInfo: AccountInfo<Buffer>) => {
+        const accountData = AccountLayout.decode(accountInfo.data);
+
+        const dividedTokenAmount = new BN(accountData.amount) / new BN(LAMPORTS_PER_SOL);
```
Comment (R-K-H) [inline contexts/BalancesContext.tsx]: This may be inaccurate but not entirely sure. The UI for USDC isn't accurate.

**Review by R-K-H — CHANGES_REQUESTED** [review]

File: contexts/BalancesContext.tsx (hunk 1 above)
Comment (LukasDeco) [reply contexts/BalancesContext.tsx]: Okay addressed this with a fix.

File: contexts/BalancesContext.tsx (hunk 2 above)
Comment (LukasDeco) [reply contexts/BalancesContext.tsx]: Fixed this in last commit. Now using specific base lots for each.

**Review by R-K-H — DISMISSED** [review]
#44 see this for revisions and updates.

**Pull request merged**